dot/parachain/backing/per_relay_parent_state.go

```diff
@@ -58,29 +58,29 @@ func (rpState *perRelayParentState) importStatement(
 		return rpState.table.importStatement(&rpState.tableContext, signedStatementWithPVD.SignedFullStatement)
 	}
 
-	// PersistedValidationData should not be nil if the statementVDT is Seconded.
-	if signedStatementWithPVD.PersistedValidationData == nil {
-		return nil, errNilPersistedValidationData
-	}
-
-	statementVDTSeconded := statementVDT.(parachaintypes.Seconded)
-	hash, err := parachaintypes.CommittedCandidateReceipt(statementVDTSeconded).Hash()
+	committedCandidateReceipt := parachaintypes.CommittedCandidateReceipt(statementVDT.(parachaintypes.Seconded))
+	candidateHash, err := parachaintypes.GetCandidateHash(committedCandidateReceipt)
 	if err != nil {
 		return nil, fmt.Errorf("getting candidate hash: %w", err)
 	}
 
-	candidateHash := parachaintypes.CandidateHash{Value: hash}
-
 	if _, ok := perCandidate[candidateHash]; ok {
 		return rpState.table.importStatement(&rpState.tableContext, signedStatementWithPVD.SignedFullStatement)
 	}
 
+	// PersistedValidationData should not be nil if the statementVDT is Seconded.
+	if signedStatementWithPVD.PersistedValidationData == nil {
+		return nil, errNilPersistedValidationData
+	}
+
```

Comment:
We need `PersistedValidationData` only to enter a candidate in `perCandidate` if we don't already have it.

So, I moved this logic down after we checked if the candidate entry is present in `perCandidate`.